src/compute/src/render.rs

```diff
@@ -824,27 +842,43 @@ where
         self.scope.clone().region_named("Main Body", |region| {
             let depends = body.depends();
             self.enter_region(region, Some(&depends))
-                .render_letfree_plan(body)
+                .render_letfree_plan(object_id, body)
                 .leave_region()
         })
     }
 
     /// Renders a plan to a differential dataflow, producing the collection of results.
     ///
     /// The plan must _not_ contain any `Let` or `LetRec` nodes.
-    fn render_letfree_plan(&mut self, plan: FlatPlan) -> CollectionBundle<G> {
-        let (mut nodes, root_id, topological_order) = plan.destruct();
+    fn render_letfree_plan(&mut self, object_id: GlobalId, plan: FlatPlan) -> CollectionBundle<G> {
+        let (mut steps, root_id, topological_order) = plan.destruct();
 
         // Rendered collections by their `LirId`.
         let mut collections = BTreeMap::new();
 
-        for id in topological_order {
-            let node = nodes.remove(&id).unwrap();
-            let mut bundle = self.render_plan_node(node, &collections);
-
-            self.log_operator_hydration(&mut bundle, id);
+        for lir_id in topological_order {
```

Comment:
We're constructing all operators in this loop, so we could change the log event to container a boxed slice of Lir mappings instead of logging each individually. I think this would make sense because it would avoid increasing the size of the log event.

Ideally we'd measure the performance impact, but I fear it'll be hard to see a difference if we only enable the feature for non-transient dataflows.